pulser/simulation/simresults.py

```diff
@@ -54,6 +59,194 @@ def __init__(self, run_output, dim, size, basis_name, meas_basis=None):
+
+        Note:
+            The results are presented using a big-endian representation,
+            according to the pre-established qubit ordering in the register.
+            This means that, when sampling a register with qubits ('q0','q1',
+            ...), in this order, the corresponding value, in binary, will be
+            0Bb0b1..., where b0 is the outcome of measuring 'q0', 'b1' of
+            measuring 'q1' and so on.
+
+        Keyword Args:
+            meas_basis (str, default=None): 'ground-rydberg' or 'digital'. If
+                left as None, uses the measurement basis defined in the
+                original sequence.
+            N_samples (int, default=1000): Number of samples to take.
+
+        Raises:
+            ValueError: If trying to sample without a defined 'meas_basis' in
+                the arguments when the original sequence is not measured.
+        """
+        pass
+
+    @abstractmethod
+    def sample_final_state(self, meas_basis=None, N_samples=1000):
+        pass
+
+
+class NoisyResults(SimulationResults):
+    """
+    Results of a noisy simulation run of a pulse sequence. Contrary to a
+    CleanResults object, this object contains a unique Counter describing the
+    state distribution at the time it was created by using
+    sim.run(spam=True, t).
+
+    Contains methods for studying the populations and extracting useful
+    information from them.
+    """
+
+    def __init__(self, run_output, size, basis_name, meas_basis, dim=2):
+        """
+        Initializes a new NoisyResults instance.
+
+        Warning : Can't have single-atom Hilbert spaces with dimension bigger
+            than 2 for NoisyResults objects.
+            This is not the case for a CleanResults object, containing states
+            in Hilbert space, but NoisyResults contains a probability
+            distribution of bitstrings, not atomic states
+
+        Args:
+            run_output (Counter) : Counter returning the probability of each
+                multi-qubits state, represented as a bitstring.
+            dim (int): The dimension of the local space of each atom (2 or 3).
+            size (int): The number of atoms in the register.
+            basis_name (str): The basis indicating the addressed atoms after
+                the pulse sequence ('ground-rydberg', 'digital' or 'all').
+
+        Keyword Args:
+            meas_basis (None or str): The basis in which a sampling measurement
+                is desired.
+        """
+        super().__init__(run_output, dim, size, basis_name,
+                         meas_basis)
+
+    @property
+    def states(self):
+        """Probability distribution of the bitstrings"""
+        return self._states
+
+    def get_final_state(self):
+        """Get the final state (density matrix here !) of the simulation.
+
+        Returns:
+            qutip.Qobj: The resulting final state as a density matrix.
```

Comment:
Not a density matrix. Please correct description.

Reply:
Right, it's just a useful trick to compute the expectation values of observables. I changed the description in the latest PR to avoid confusions about what this object is - the diagonal elements of the density matrix.